### ISSUE TITLE
Jbuilder produces different library filenames than the replaced build system

### ISSUE THREAD

**Issue by aantron** [ocsigen/lwt#456]
- Library filenames were hyphenated. They are now underscored.
- The PPX files have been renamed. This is probably not important.
- We don't generate `.annot` files anymore. I'm assuming this doesn't matter at all, but not fully sure.

Eventually, we want to let Jbuilder take care of this internally, and not worry about it. That means we will switch to underscored names, etc., and that will be a breaking change in 4.0.0.

For now, I think we need to doctor the installation directory to avoid these changes. By 3.2.0, we should figure out if we will hard-break this, or provide a transitional state.

See https://github.com/ocaml/opam-repository/pull/9881.

cc @andrewray FYI. I'll work on this.

This delays https://github.com/ocaml/opam-repository/pull/9876. cc @avsm.

See the diff of ``ls `ocamlfind query lwt` `` between Lwt 3.0.0 and tag `3.1.0` (which is about to get adjusted):

```diff
--- old	2017-07-20 08:21:14.000000000 -0500
+++ new	2017-07-20 08:23:25.000000000 -0500
@@ -1,33 +1,33 @@
 META
-liblwt-unix_stubs.a
-lwt-log.a
-lwt-log.cma
-lwt-log.cmxa
-lwt-log.cmxs
-lwt-preemptive.a
-lwt-preemptive.cma
-lwt-preemptive.cmxa
-lwt-preemptive.cmxs
-lwt-simple-top.a
-lwt-simple-top.cma
-lwt-simple-top.cmxa
-lwt-simple-top.cmxs
-lwt-syntax-log.a
-lwt-syntax-log.cma
-lwt-syntax-log.cmxa
-lwt-syntax-log.cmxs
-lwt-syntax-options.a
-lwt-syntax-options.cma
-lwt-syntax-options.cmxa
-lwt-syntax-options.cmxs
-lwt-syntax.a
-lwt-syntax.cma
-lwt-syntax.cmxa
-lwt-syntax.cmxs
-lwt-unix.a
-lwt-unix.cma
-lwt-unix.cmxa
-lwt-unix.cmxs
+liblwt_unix_stubs.a
+lwt_log.a
+lwt_log.cma
+lwt_log.cmxa
+lwt_log.cmxs
+lwt_preemptive.a
+lwt_preemptive.cma
+lwt_preemptive.cmxa
+lwt_preemptive.cmxs
+lwt_simple_top.a
+lwt_simple_top.cma
+lwt_simple_top.cmxa
+lwt_simple_top.cmxs
+lwt_syntax_log.a
+lwt_syntax_log.cma
+lwt_syntax_log.cmxa
+lwt_syntax_log.cmxs
+lwt_syntax_options.a
+lwt_syntax_options.cma
+lwt_syntax_options.cmxa
+lwt_syntax_options.cmxs
+lwt_syntax.a
+lwt_syntax.cma
+lwt_syntax.cmxa
+lwt_syntax.cmxs
+lwt_unix.a
+lwt_unix.cma
+lwt_unix.cmxa
+lwt_unix.cmxs
 lwt.a
 lwt.cma
 lwt.cmi
@@ -52,8 +52,11 @@
 lwt_condition.cmti
 lwt_condition.cmx
 lwt_condition.mli
+lwt_config.cmi
+lwt_config.cmt
 lwt_config.cmx
 lwt_config.h
+lwt_config.ml
 lwt_daemon.cmi
 lwt_daemon.cmt
 lwt_daemon.cmti
@@ -139,7 +142,10 @@
 lwt_sequence.cmti
 lwt_sequence.cmx
 lwt_sequence.mli
+lwt_simple_top.cmi
+lwt_simple_top.cmt
 lwt_simple_top.cmx
+lwt_simple_top.ml
 lwt_stream.cmi
 lwt_stream.cmt
 lwt_stream.cmti
@@ -170,7 +176,13 @@
 lwt_unix.cmti
 lwt_unix.cmx
 lwt_unix.h
+lwt_unix.mli
+lwt_unix_jobs_generated.cmi
+lwt_unix_jobs_generated.cmt
 lwt_unix_jobs_generated.cmx
+lwt_unix_jobs_generated.ml
+lwt_unix_unix.h
+lwt_unix_windows.h
 opam
 opam.config
 pa_lwt.cmi
@@ -183,43 +195,17 @@
 pa_lwt_log.cmti
 pa_lwt_log.cmx
 pa_lwt_log.mli
+pa_lwt_options.cmi
+pa_lwt_options.cmt
 pa_lwt_options.cmx
-ppx.a
-ppx.cma
-ppx.cmxa
-ppx.cmxs
+pa_lwt_options.ml
+ppx.exe
+ppx_lwt.a
+ppx_lwt.cma
 ppx_lwt.cmi
 ppx_lwt.cmt
 ppx_lwt.cmti
 ppx_lwt.cmx
+ppx_lwt.cmxa
+ppx_lwt.cmxs
 ppx_lwt.mli
-lwt.annot
-lwt_bytes.annot
-lwt_chan.annot
-lwt_condition.annot
-lwt_daemon.annot
-lwt_engine.annot
-lwt_gc.annot
-lwt_io.annot
-lwt_list.annot
-lwt_log.annot
-lwt_log_core.annot
-lwt_log_rules.annot
-lwt_main.annot
-lwt_mutex.annot
-lwt_mvar.annot
-lwt_pool.annot
-lwt_pqueue.annot
-lwt_preemptive.annot
-lwt_process.annot
-lwt_result.annot
-lwt_sequence.annot
-lwt_stream.annot
-lwt_switch.annot
-lwt_sys.annot
-lwt_throttle.annot
-lwt_timeout.annot
-lwt_unix.annot
-pa_lwt.annot
-pa_lwt_log.annot
-ppx_lwt.annot
```

**Comment by aantron** [comment]
After looking into this, I decided not to fix it. I was able to get almost all the libraries hyphenated, but for the stubs files (including `dlllwt_unix_stubs.so`), it seems there is no easy way to get them hyphenated short of renaming the source files.

The greatest actual risk of breakage is probably from calling the PPX binary directly, and expecting it to be in your switch's `bin/`. It is now installed `lib/`.

However, we haven't documented any way to use Lwt that does not involve ocamlfind or Jbuilder in a very long time. I hope that means we can declare the directory layout not to be part of the public interface of the Lwt packaging.
